reconcile/saas_auto_promotions_manager/utils/saas_files_inventory.py

```diff
@@ -34,6 +34,11 @@ def __init__(self, saas_files: Iterable[SaasFile]):
         self._assemble_publishers()
         self._remove_unsupported()
 
+
+    @property
+    def publishers_with_subscribers(self) -> list[Publisher]:
```

Comment:
@fishi0x01 note that I added this to keep tests adequate.

This changes things then, and I'm not sure I understand the consequences. Before this PR `publishers` report nodes that have subscribers. After this PR is `publishers` will report all nodes that publish, and you need to specifically access `publishers_with_subscribers` if you want to get to the previous behaviour.

Alternatively we could reverse it, i.e. keep `publishers` as it was, add `publishers_without_subscribers`, and leverage that one in the S3 exporter.

What do you prefer?